### PR TITLE
added snippet_utils.py

### PR DESCRIPTION
**Pull request by KirtiPratihar**
This Python code provides utility functions for managing code snippets stored in a "code_snippets" directory. The script includes three key functionalities:

1.**display(snippet_name):**- Reads and prints the content of a Python file (a code snippet) in the console.
The snippet file is located in a subfolder called "code_snippets", and its name is passed as an argument (snippet_name). 

2.**copy_to_clipboard(snippet_name):**-Copies the content of the specified snippet file to the system clipboard.
The function supports different platforms:
For Linux: It uses the xclip command.
For Windows: It uses the clip.exe command.
Other platforms are not supported, and it raises an OSError for unsupported operating systems.

3.**write_to_root(snippet_name):**-Writes the content of the snippet file to the root directory where the script is executed.
The file is saved with the same name as the snippet but in the current working directory instead of the "code_snippets" folder.

### CODE REVIEW

File: snippet_utils.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+# from package_name.snippet_utils import display, copy_to_clipboard, write_to_root
+
+# def main():
+#     filename = 'example'  # replace with your filename
+#     display(filename)
+#     copy_to_clipboard(filename)
+#     write_to_root(filename)
+
+# if __name__ == "__main__":
+#     main()
```
Comment (jindalpriyanshu101) [inline snippet_utils.py:88]: Why?

**Comment by jindalpriyanshu101** [comment]
I don't understand the context? what `code_snippets` directory are you talking about? and why you created this `snippet_utils.py`

Also, do link the issue which you're solving.

**Comment by KirtiPratihar** [comment]
this is the issue which was assigned- **https://github.com/AmateursLeague/sneaky-package/issues/21** and as solution, I have made **snippet_utils.py** , so that we can asscess everything from one .

In simple terms, instead of calling 3 features separately, we’ll call the methods separately from one file.

**Comment by Wahid7852** [comment]
Hi Kirti, can you update your code? We have support for macOS as well and updated functionality in the files, so please do the necessary.

File: snippet_utils.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    snippet_path = os.path.join(
+        os.path.dirname(__file__), "code_snippets", f"{snippet_name}.py"
+    )
+
+    with open(snippet_path, "r") as file:
+        source_code = file.read()
+        print(source_code)
+
+
+def copy_to_clipboard(snippet_name):
```
Comment (this-is-yaash) [inline snippet_utils.py:21]:
```suggestion
def graph(snippet_name):
```

File: snippet_utils.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            ["C:\\Windows\\System32\\clip.exe"],
+            input=source_code.strip().encode(),
+            check=True,
+        )
+
+    else:
+        raise OSError("Unsupported operating system")
+
+
+def write_to_root(snippet_name):
```
Comment (this-is-yaash) [inline snippet_utils.py:54]:
```suggestion
def piechart(snippet_name):
```

File: snippet_utils.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
+import os
+import subprocess
+import sys
+
+def display(snippet_name):
```
Comment (this-is-yaash) [inline snippet_utils.py:5]:
```suggestion
def models(snippet_name):
```

File: snippet_utils.py (hunk 1 above)
Comment (this-is-yaash) [inline snippet_utils.py:88]: Remove these lines as well!

**Review by this-is-yaash — COMMENTED** [review]
I have noted 6 changes, please make necessary updates!
Additionally change the filename to `display.py`

File: snippet_utils.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        raise OSError("Unsupported operating system")
+
+
+def piechart(snippet_name):
+    """
+    Writes the file to the root directory of execution.
+    
+    Args:
+        snippet_name (str): The name of the snippet to write.
+    """
```
Comment (jindalpriyanshu101) [inline snippet_utils.py:49]: Remove these comments

**Review by jindalpriyanshu101 — CHANGES_REQUESTED** [review]

**Review by jindalpriyanshu101 — APPROVED** [review]
LGTM, Approved as requested changes are implemented.

**Review by this-is-yaash — APPROVED** [review]
LGTM!
Good job!

**Comment by this-is-yaash** [comment]
@KirtiPratihar Star⭐ this repo if you like it!
Thanks your contributions!

**Pull request merged**